src/components/ItemList/Item.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+
+function Item({ item, onChange, onDelete }) {
+  const [isEditing, setIsEditing] = useState(false);
+  let itemContent;
+  if (isEditing) {
+    itemContent = (
+      <>
+        <input
+          value={item.text}
+          onChange={(e) => {
+            onChange({
+              ...item,
+              text: e.target.value,
+            });
+          }}
+        />
+
+        <button type="button" onClick={() => setIsEditing(false)}>
```

Comment:
on this onClick should be a onEdit function does what the current onChange function is doing and setting the edit mode to false. 
something like: 

const onEdit(task) {
    dispatch({
      type: 'changed',
      task,
    })
    setIsEditing(false)
  };